std/array.d

```diff
@@ -1586,25 +1635,32 @@ ElementEncodingType!(ElementType!RoR)[] join(RoR, R)(RoR ror, R sep)
     static if(hasCheapIteration!RoR && (hasLength!RoRElem || isNarrowString!RoRElem))
     {
         import std.conv : emplaceRef;
-        size_t length;
+        size_t length;          // length of result array
+        size_t rorLength;       // length of range ror
         foreach(r; ror.save)
-            length += r.length + sepArr.length;
-        length -= sepArr.length;
-        auto result = uninitializedArray!(RetTypeElement[])(length);
+        {
+            length += r.length;
+            ++rorLength;
+        }
+        if (!rorLength)
+            return null;
+        length += (rorLength - 1) * sepArr.length;
+
+        import std.conv : trusted;
+        auto result = trusted!(() => uninitializedArray!(RetTypeElement[])(length));
```

Comment:
Isn't this more or less the logic that std.array.array provides, already with a safe interface? Would it be pertinent to call that instead?